app.js

```diff
@@ -1,20 +1,41 @@
 // require some stuff
 var fs = require('fs');
+var https = require('https');
 var path = require('path');
 var WARCStream = require('warc');
-var dataFolder = './data/';
-// read filenames inside the specified folder
-fs.readdir(dataFolder, function (err, files) {
-    files.forEach(function (file) {
-        // we only want none compressed .wet, .wat or .warc files
-        if (path.extname(file).match(/\.wet|\.wat|\.warc/)) {
-            // open each file in the folder as stream and pipe it to the warc parser
-            var WARCParser = new WARCStream();
-            fs.createReadStream(dataFolder + file).pipe(WARCParser).on('data', function (data) {
-                // log content of each entry in console
-                var content = data.content.toString('utf8');
-                console.log(content);
-            });
-        }
-    });
-});
+// download web archive file
+function downloadFile(fileURL, directory) {
+    // create new directory
+    try {
+        fs.mkdirSync(directory);
+    }
+    catch (e) {
+        // catch exception if directory already exists
+        if (e.code != 'EEXIST')
+            throw e;
+    }
+    // extract filename from url
+    var filename = path.basename(fileURL);
+    // check if file already exists
+    if (!fs.exists(directory + filename)) {
+        // create a new file on disk
+        var file_1 = fs.createWriteStream(directory + filename);
+        // download file
+        https.get(fileURL, function (response) { response.pipe(file_1); });
+    }
+}
+// digest web archive file
+function digestFile(filepath) {
+    // we only want none compressed .wet, .wat or .warc files
+    if (path.extname(filepath).match(/\.wet|\.wat|\.warc/)) {
+        // open each file in the folder as stream and pipe it to the warc parser
+        var WARCParser = new WARCStream();
+        fs.createReadStream(filepath).pipe(WARCParser).on('data', function (data) {
+            // log content of each entry in console
+            var content = data.content.toString('utf8');
+            console.log(content);
```

Comment:
Just a hint: do not commit logging that is solely for your debugging. It only makes it harder for your teammates to debug their code if their console is full of messages they don't understand.